code/game/objects/structures/railing.dm

```diff
@@ -278,35 +278,49 @@
 /obj/structure/railing/ex_act(severity)
 	qdel(src)
 
-/obj/structure/railing/can_climb(var/mob/living/user, climb_dir, post_climb_check=0)
+/obj/structure/railing/can_climb(var/mob/living/user, post_climb_check=0)
 	. = ..()
-	var/turf/dest = get_step(user, climb_dir)
-	if(. && get_turf(user) == get_turf(src))
-		if(turf_is_crowded(TRUE) || !user.Adjacent(dest) || turf_contains_dense_objects(dest) && get_turf(src) != dest)
+	if(!.)
+		return
+
+	var/turf/destination_turf = get_destination_turf(user)
+
+	if(destination_turf.density)
+		to_chat(user, SPAN_DANGER("You can't climb into \the [destination_turf]."))
+		return FALSE
+
+	if(!Adjacent(destination_turf))
+		to_chat(user, SPAN_DANGER("You can't climb there, the way is blocked."))
+		return FALSE
+
+	for(var/atom/A in destination_turf.contents - src)
+		if(A.density && !(A.flags & ON_BORDER))
```

Comment:
This appears to trigger if the turf is shared with another railing, even if it faces another direction.